src/libraries/System.Private.CoreLib/src/System/Numerics/INumberBase.cs

```diff
@@ -61,16 +64,50 @@ public interface INumberBase<TSelf>
         static abstract TSelf CreateTruncating<TOther>(TOther value)
             where TOther : INumberBase<TOther>;
 
+        /// <summary>Determines if a value is in its canonical representation.</summary>
+        /// <param name="value">The value to be checked.</param>
+        /// <returns><c>true</c> if <paramref name="value" /> is in its canonical representation; otherwise, <c>false</c>.</returns>
+        static abstract bool IsCanonical(TSelf value);
+
+        /// <summary>Determines if a value represents a complex value.</summary>
+        /// <param name="value">The value to be checked.</param>
+        /// <returns><c>true</c> if <paramref name="value" /> is a complex number; otherwise, <c>false</c>.</returns>
+        /// <remarks>This function returns <c>false</c> for a complex number <c>a + bi</c> where <c>b</c> is zero.</remarks>
```

Comment:
> The implementation doesn't match this description, as you also return false if a is zero.

This is likely a copy/paste issue.

The premise is that a `complex` number has both a `real` and `imaginary` part. That is, it is represented as `a + bi`. However, it's not useful as an API to only return "true" since all numbers are representable (at least in principle) as some complex number. This leads to 3 concepts: `IsReal`, `IsImaginary`, and `IsComplex`

A `real` number is some number that only has a `real` part. That is, its `imaginary` part is `0`.
An `imaginary` number is some number that only has an `imaginary` part. That is, its `real` part is `0`
A `complex` number is some number that has both a `real` and `imaginary` parts. That is, neither its `real` or `imaginary` parts are `0`

This quantifies `a + 0i` as `real` and `0 + bi` as `imaginary`. This also means `0` is both real and imaginary (which it is) and that all other values are `complex`

Reply:
This allows users to special-case `pure real`, `pure imaginary`, and `true complex` values as appropriate in their API surface.

These are often the three scenarios that you have to specialize within `Complex` itself and more complicated algorithms.

Reply:
That makes sense. As long as this behavior is properly documented, I think it should be fine.

Reply:
Going to log an issue tracking updating this as a post cleanup item to avoid spinning CI again.

Reply:
https://github.com/dotnet/runtime/issues/69694